pyaarlo/tfa.py

```diff
@@ -48,110 +48,126 @@ def start(self):
+                # If we find the code in one of the mails then save it, otherwise save None
+                mail_id = next((x for x in recent_ids if self.is_arlo_mail(x) and self.get_code(x) is not None), None)
+
+                # If we found the code then return it and delete the mail
+                # We should not use unnecessary space in the inbox
+                if mail_id is not None:
+                    code = self.get_code(mail_id)
+                    self._arlo.debug("2fa-imap: code={}".format(code))
+                    self.delete_mail(mail_id)
+                    return code
+
+                # Set the self._old_ids to the mail ids we just searched
+                self._arlo.debug("2fa-imap: Setting old value")
+                self._old_ids = recent_ids
 
-                # new message...
-                old_ids = self._old_ids[0].split()
-                for msg_id in self._new_ids[0].split():
-
-                    # seen it?
-                    if msg_id in old_ids:
-                        continue
-
-                    # New message. Look at all the parts and try to grab the code, if we catch an exception
-                    # just move onto the next part.
-                    self._arlo.debug("2fa-imap: new-msg={}".format(msg_id))
-                    res, msg = self._imap.fetch(msg_id, "(BODY[TEXT])")
-                    if isinstance(msg[0][1], bytes):
-                        for part in email.message_from_bytes(msg[0][1]).walk():
-                            try:
-                                for line in part.get_payload(decode=True).splitlines():
-                                    # match code in email, this might need some work if the email changes
-                                    code = re.match(r"^\W*(\d{6})\W*$", line.decode())
-                                    if code is not None:
-                                        self._arlo.debug(
-                                            "2fa-imap: code={}".format(code.group(1))
-                                        )
-                                        return code.group(1)
-                            except:
-                                self._arlo.debug("trying next part")
-
-                # update old so we don't keep trying new
-                self._old_ids = self._new_ids
-
-            # problem parsing the message, force a fail
+            # Problem parsing the message, force a fail
             except Exception as e:
                 self._arlo.error(f"imap message read failed{str(e)}")
                 return None
 
         return None
 
+    # Fetch the mail and look at the sender, if Arlo the return True, otherwise False
+    def is_arlo_mail(self, mail_id):
+        self._arlo.debug("2fa-imap: Checking if mail is Arlo mail, mail id: " + format(mail_id))
+        res, msg = self._imap.fetch(mail_id, "(RFC822)")
+        if isinstance(msg[0][1], bytes):
```

Comment:
For some reason, I have to use msg[1][1] with my yahoo mail. Mentioning it in #96. No idea why it works differently if you've also tested with a yahoo mail

Reply:
I made a printout of msg[0][1] and msg[0][1]..

msg[0][1]: 50

... nothing else...

msg[1][1]:
b'Received: from 127.0.0.1\r\n by atlas-production.v2-mail-prod1-gq1.omega.yahoo.com pod-id atlas--production-gq1-6db95f6ddc-xz57w.gq1.yahoo.com with HTTP..... + a lot more. LOOOOOONG string containing the entire mail